nats.go

```diff
@@ -3162,6 +3156,18 @@ func (s *Subscription) validateNextMsgState() error {
 	return nil
 }
 
+// This is called when the sync channel has been closed.
+// The error returned will be either connection or subscription
+// closed depending on what caused NextMsg() to fail.
+func (s *Subscription) getNextMsgErr() error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
This is in fast (kinda) path, can we not use defer?

Reply:
I was going to, but realize that this is when the subscription (or connection) is closed. I would not consider this fast path.

Reply:
Meaning that this is a single event per subscription. After that call, subscription is dead.

Reply:
Good point.